src/rust/catloop/mod.rs

```diff
@@ -778,29 +752,33 @@ impl CatloopLibOS {
         }
         qt.into()
     }
+
+    /// This function inserts `coroutine` into the scheduler with [task_id] to identify it.
+    fn insert_coroutine(
+        scheduler: &mut Scheduler,
+        task_name: String,
+        coroutine: Pin<Box<Operation>>,
+    ) -> Result<TaskHandle, Fail> {
+        let task: OperationTask = OperationTask::new(task_name, coroutine);
+        match scheduler.insert(task) {
+            Some(handle) => Ok(handle),
+            None => Err(Fail::new(libc::EAGAIN, "cannot schedule coroutine")),
+        }
+    }
+
+    /// Removes the coroutine associated with `handle` from the underlying scheduler.
+    fn remove_coroutine(scheduler: &mut Scheduler, handle: &TaskHandle) -> Option<Box<dyn Task>> {
+        scheduler.remove(handle)
+    }
 }
 
 //======================================================================================================================
 // Trait Implementations
 //======================================================================================================================
 
 impl Drop for CatloopLibOS {
-    // Releases all sockets allocated by Catnap.
     fn drop(&mut self) {
-        for (_, queue) in self.qtable.borrow().get_values() {
-            if let Some(duplex_pipe) = queue.get_pipe() {
-                if duplex_pipe.close().is_err() {
-                    warn!("drop(): failed to close duplex pipe");
-                }
-            }
-            if let Socket::Active(Some(addr)) | Socket::Passive(addr) = queue.get_socket() {
-                if EphemeralPorts::is_private(addr.port()) {
-                    if self.ephemeral_ports.borrow_mut().free(addr.port()).is_err() {
-                        warn!("close(): leaking ephemeral port (port={})", addr.port());
-                    }
-                }
-            }
-        }
+        // This is a stateless object. All resources are managed by underlying components.
```

Comment:
Do we need to implement the `drop()` if this struct is stateless?

Reply:
Removed!